### ISSUE TITLE
STY/TST: Usage of pysat-generated files in unit tests

### ISSUE THREAD

**Issue by jklenzing** [pysat/pysat#960]
**Is your feature request related to a problem? Please describe.**
Currently, the test instruments create fake lists of files to generate fake data to test.  Some of the file list routines here and elsewhere need to create fake empty files for testing.  Currently, this is done by generating fake files manually.  Because the "data" directories for these instruments are not created, `utils.testing.pre_dir` is required to generate the path before empty files can be created.

**Describe the solution you'd like**
It would be useful if most of this infrastructure could be imported directly from pysat to test the `list_files` routines in external packages.  Possibilities for streamlining
- Add a tag to `pysat_testing` which would generate fake files as part of the download routine.  This would be excluded from end-to-end testing in test_instruments.py via appropriate flags.  
- Have the temporary directory structure as part of the instrument, or perhaps as an importable class structure similar to test_instruments.py
- Attach the new list_files routine to the test instrument upon loading

**Describe alternatives you've considered**
Rather than exporting code from the core package, write up philosophy of tests in the documentation to assist in custom codes.  Probably need a philosophy of testing section either way, but this would rely on documentation rather than code to solve the problem.

**Additional context**
Based on discussions in pysat/pysatMadrigal#60

**Comment by aburrell** [comment]
> Add a tag to pysat_testing which would generate fake files as part of the download routine. This would be excluded from end-to-end testing in test_instruments.py via appropriate flags.

I don't think this is the way to go, because it won't necessarily create the desired type of file for testing

> Have the temporary directory structure as part of the instrument, or perhaps as an importable class structure similar to test_instruments.py

Something like this perhaps?  Maybe add a `create_dir_struct` kwarg to the test Instrument load routine.  Then there would be a place to put any files that need to be created for instrument module tests.  The "hard" part would be making the directory clean-up call a part of the Instrument clean up.

> Attach the new list_files routine to the test instrument upon loading

This wouldn't solve the pysatMadrigal issue, so I don't think it's a good solution.

> Rather than exporting code from the core package, write up philosophy of tests in the documentation to assist in custom codes

I like this option the best.  We (will) have the routines needed, and just need to make sure people writing tests remember that the list files doesn't create actual files or directories, so those need to be created and cleaned in the test class setup/teardown.

**Comment by jklenzing** [comment]
Code here could be used as part of #779
